eth/client_ticketbroker.go

```diff
@@ -26,7 +26,7 @@ func (c *client) FundDeposit(amount *big.Int) (*types.Transaction, error) {
 	opts := c.transactOpts()
 	opts.Value = amount
```

Comment:
This was already here before this PR, but since `transactOpts` is stored on `client` and `c.transactOpts()` returns a pointer to that field does the update to the `Value` field in the next line get reflected in that field such that the next call to `c.transactOpts()` would return a pointer to a `bind.TransactOpts` with the `Value` field set (which we wouldn't want)?

Reply:
Actually, that's not the case, because `c.transactOpts()` returns a copy of `TransactOpts`. The copying happens [here](https://github.com/leszko/go-livepeer/blob/cdfb8d15dc15300483eaed64a2e8f4d9bf7b796e/eth/client.go#L368).

Reply:
Aha I see.

The field `c.transactOpts` is a value so `opts := c.transactOpts` will copy the value and assign it to `opts`. Then, `c.transactOpts()` will return a pointer to the *copy* of `c.transactOpts` and not to `c.transactOpts`. As a result, any update to the pointer returned by `c.transactOpts()` will only impact the copy of `c.transactOpts` and not `c.transactOpts`. As a result, the update to the `Value` field in the referenced line will only affect a copy of `c.transactOpts` and since we call `c.transactOpts()` to return a pointer to a fresh copy of `c.transactOpts` for each method we don't need to worry about that update affecting other method calls.

Reply:
yes, exactly. Also, the reason for copying `TransactOpts` makes the code thread-safe.